### PR TITLE
[Merged by Bors] - chore: refactor to `List.get`

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Analytic/Composition.lean [hunk 1]
```diff
@@ -1033,7 +1024,7 @@ def sigmaCompositionAux (a : Composition n) (b : Composition a.length)
       (by
         rw [← a.blocks.join_splitWrtComposition b]
         exact mem_join_of_mem (List.get_mem _ _ _) hi)
-  blocks_sum := by simp only [Composition.blocksFun, nthLe_map', Composition.gather]; rfl
+  blocks_sum := by simp only [Composition.blocksFun, get_map, Composition.gather]
 #align composition.sigma_composition_aux Composition.sigmaCompositionAux
 
 -- porting note: this needs `Composition.blocksFun` to be refactored in order to remove `nthLe`
```
Comment (eric-wieser) [inline Mathlib/Analysis/Analytic/Composition.lean:1030]: You've now done this refactor, right? Can the `nthLe` be easily removed?

File: Mathlib/Combinatorics/Composition.lean [hunk 2]
```diff
@@ -149,45 +149,35 @@ theorem blocks_length : c.blocks.length = c.length :=
   rfl
 #align composition.blocks_length Composition.blocks_length
 
--- porting note: TODO, refactor to `List.get`
-set_option linter.deprecated false in
 /-- The blocks of a composition, seen as a function on `Fin c.length`. When composing analytic
 functions using compositions, this is the main player. -/
-def blocksFun : Fin c.length → ℕ := fun i => nthLe c.blocks i i.2
+def blocksFun : Fin c.length → ℕ := fun i => c.blocks.get i
```
Comment (eric-wieser) [inline Mathlib/Combinatorics/Composition.lean:154]: Untested
```suggestion
def blocksFun : Fin c.length → ℕ := c.blocks.get
```

File: Mathlib/Combinatorics/Composition.lean [hunk 3]
```diff
@@ -955,9 +936,7 @@ theorem blocks_partial_sum {i : ℕ} (h : i < c.boundaries.card) :
     simp [blocks, Nat.lt_of_succ_lt_succ h]
   have B : i < c.boundaries.card := lt_of_lt_of_le A (by simp [blocks, length, Nat.sub_le])
   rw [sum_take_succ _ _ A, IH B]
-  simp only [blocks, blocksFun, nthLe_ofFn']
-  apply add_tsub_cancel_of_le
-  simp
+  simp  [blocks, blocksFun, get_ofFn]
```
Comment (eric-wieser) [inline Mathlib/Combinatorics/Composition.lean:939]:
```suggestion
  simp [blocks, blocksFun, get_ofFn]
```

File: Mathlib/Analysis/Analytic/Composition.lean (hunk 1 above)
Comment (mo271) [reply Mathlib/Analysis/Analytic/Composition.lean]: Yeah, the `nthLe` could easily removed in all cases, but in `sizeUpTo_sizeUpTo_add`, where I didn't yet find the `get`-version of `List.sum_take_succ`, which introduces the `nthLe` there...
I will have another look.

Comment (mo271) [reply Mathlib/Analysis/Analytic/Composition.lean]: Changing `List.sum_take_succ` (and therefore `List.prod_take_succ`), will require some more change, so I propose doing this in #8043. So this pull request here is ready for review...

File: Mathlib/Analysis/Analytic/Composition.lean [hunk 4]
```diff
@@ -596,15 +596,15 @@ theorem compChangeOfVariables_length (m M N : ℕ) {i : Σ n, Fin n → ℕ}
...
 theorem compChangeOfVariables_blocksFun (m M N : ℕ) {i : Σ n, Fin n → ℕ}
     (hi : i ∈ compPartialSumSource m M N) (j : Fin i.1) :
     (compChangeOfVariables m M N i hi).2.blocksFun
         ⟨j, (compChangeOfVariables_length m M N hi).symm ▸ j.2⟩ =
       i.2 j := by
   rcases i with ⟨n, f⟩
   dsimp [Composition.blocksFun, Composition.blocks, compChangeOfVariables]
-  simp only [map_ofFn, List.nthLe_ofFn, Function.comp_apply]
+  simp only [map_ofFn, List.get_ofFn, Function.comp_apply]
+  rfl
```
Comment (eric-wieser) [inline Mathlib/Analysis/Analytic/Composition.lean:607]: Is there a lemma missing here?

File: Mathlib/Combinatorics/Composition.lean [hunk 5]
```diff
@@ -232,9 +222,10 @@ theorem sizeUpTo_le (i : ℕ) : c.sizeUpTo i ≤ n := by
 #align composition.size_up_to_le Composition.sizeUpTo_le
 
 theorem sizeUpTo_succ {i : ℕ} (h : i < c.length) :
-    c.sizeUpTo (i + 1) = c.sizeUpTo i + c.blocks.nthLe i h := by
+    c.sizeUpTo (i + 1) = c.sizeUpTo i + c.blocks.get ⟨i, h⟩ := by
   simp only [sizeUpTo]
   rw [sum_take_succ _ _ h]
+  rfl
```
Comment (eric-wieser) [inline Mathlib/Combinatorics/Composition.lean:228]: This looks like a missing lemma too?

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks! There are a few new trailing `rfl`s that seem to indicate a missing lemma; if it's a lemma that still needs to have `nthLe` turned to `get`, can you add a TODO comment?

File: Mathlib/Combinatorics/Composition.lean (hunk 5 above)
Comment (mo271) [reply Mathlib/Combinatorics/Composition.lean:229]: This should resolve after adapting `sum_tage_succ`. 
I added a porting note.

File: Mathlib/Analysis/Analytic/Composition.lean (hunk 4 above)
Comment (mo271) [reply Mathlib/Analysis/Analytic/Composition.lean:608]: Here, I couldn't find the missing lemma, but left a porting note stating that the was not needed before.

**Pull request closed without merging**